homeassistant/components/scrape/sensor.py

```diff
@@ -75,37 +74,47 @@ async def async_setup_platform(
 ) -> None:
     """Set up the Web scrape sensor."""
-    name: str = config[CONF_NAME]
-    resource: str = config[CONF_RESOURCE]
-    method: str = "GET"
-    payload: str | None = None
-    headers: str | None = config.get(CONF_HEADERS)
-    verify_ssl: bool = config[CONF_VERIFY_SSL]
-    select: str | None = config.get(CONF_SELECT)
-    attr: str | None = config.get(CONF_ATTR)
-    index: int = config[CONF_INDEX]
-    unit: str | None = config.get(CONF_UNIT_OF_MEASUREMENT)
-    device_class: str | None = config.get(CONF_DEVICE_CLASS)
-    state_class: str | None = config.get(CONF_STATE_CLASS)
-    username: str | None = config.get(CONF_USERNAME)
-    password: str | None = config.get(CONF_PASSWORD)
-    value_template: Template | None = config.get(CONF_VALUE_TEMPLATE)
+    _LOGGER.warning(
+        # Config flow added in Home Assistant Core 2022.7, remove import flow in 2022.9
+        "Loading Scrape via platform setup has been deprecated in Home Assistant 2022.7 "
+        "Your configuration has been automatically imported and you can "
+        "remove it from your configuration.yaml"
+    )
 
-    if value_template is not None:
-        value_template.hass = hass
+    if config.get(CONF_VALUE_TEMPLATE):
+        template: Template = Template(config[CONF_VALUE_TEMPLATE])
+        template.ensure_valid()
+        config[CONF_VALUE_TEMPLATE] = template.template
+
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_IMPORT},
+            data=config,
+        )
+    )
 
-    auth: httpx.DigestAuth | tuple[str, str] | None = None
-    if username and password:
-        if config.get(CONF_AUTHENTICATION) == HTTP_DIGEST_AUTHENTICATION:
-            auth = httpx.DigestAuth(username, password)
-        else:
-            auth = (username, password)
 
-    rest = RestData(hass, method, resource, auth, headers, None, payload, verify_ssl)
-    await rest.async_update()
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the Scrape sensor entry."""
+    name: str = entry.options[CONF_NAME]
+    resource: str = entry.options[CONF_RESOURCE]
+    select: str | None = entry.options.get(CONF_SELECT)
+    attr: str | None = entry.options.get(CONF_ATTRIBUTE)
+    index: int = int(entry.options[CONF_INDEX])
+    unit: str | None = entry.options.get(CONF_UNIT_OF_MEASUREMENT)
+    device_class: str | None = entry.options.get(CONF_DEVICE_CLASS)
+    state_class: str | None = entry.options.get(CONF_STATE_CLASS)
+    value_template: str | None = entry.options.get(CONF_VALUE_TEMPLATE)
+    entry_id: str = entry.entry_id
+
+    val_template: Template | None = None
+    if value_template is not None:
+        val_template = Template(value_template, hass)
 
-    if rest.data is None:
-        raise PlatformNotReady
+    rest = hass.data.setdefault(DOMAIN, {})[entry.entry_id]
```

Comment:
We don't need to use dict.setdefault here. We set hass.data in `async_setup_entry` in the init module.